test/langtools/tools/javac/platform/createsymbols/CreateSymbolsTestImpl.java

```diff
@@ -470,12 +481,105 @@ void testIncluded() throws Exception {
                        "t.PPI");
     }
 
+    @Test
+    void testRecords() throws Exception {
+        doPrintElementTest("package t;" +
+                           "public class T {" +
+                           "    public record R(int i, java.util.List<String> l) { }" +
+                           "}",
+                           "package t;" +
+                           "public class T {" +
+                           "    public record R(@Ann int i, long j, java.util.List<String> l) { }" +
+                           "    public @interface Ann {} " +
+                           "}",
+                           "t.T$R",
+                           """
+                           \n\
+                           public static record R(int i, java.util.List<java.lang.String> l) {
+                           \n\
+                             public R(int i,
+                               java.util.List<java.lang.String> l);
+                           \n\
+                             public final java.lang.String toString();
+                           \n\
+                             public final int hashCode();
+                           \n\
+                             public final boolean equals(java.lang.Object arg0);
+                           \n\
+                             public int i();
+                           \n\
+                             public java.util.List<java.lang.String> l();
+                           }
+                           """,
```

Comment:
There is a combination of factors here:
-jcheck (AFAIK) does not allow trailing whitespaces, even not on otherwise empty lines inside textblocks
-textblocks only remove indentation that is common on all lines.

So, without having '\n\', we would have to strip all the whitespace on the empty lines (to pass jcheck), which would mean the text block's content would no longer match the output. There are a few ways to solve this (almost surely an incomplete list):
-do some trick to have the common indent, but no trailing whitespace. '\n\' is one of them.
-not indent the text block
-do some post-processing on the text block's value or the actual test output, to make them match
-not use textblocks

Reply:
I stand corrected here - blank lines do not count when the common indent is computed. Removed here:
https://github.com/openjdk/jdk/pull/1480/commits/3aaaf28c23ddda71c77ca9923e02e5f3502cde3b